### ISSUE TITLE
Overlapping overload not reported when parameter type includes tuples with overlapping unions

### ISSUE THREAD

**Issue by insilications** [microsoft/pyright#6845]
This code should produce an "overlapping override" error, but it doesn't. Mypy currently flags this condition.

```python
from typing import overload

class A:
    @overload
    def select(self, p1: tuple[str | tuple[int | float, str], str], /) -> str: ...
    @overload
    # def select(self, p1: tuple[str | tuple[int | float, str] | int | float, str | int | float], /) -> int: ...
    def select(self, p1: tuple[tuple[int | float, str] | int | float, str | int | float], /) -> int: ...
```

**Comment by erictraut** [comment]
This is a duplicate of #6825, which I opened recently.
